src/components/application_manager/rpc_plugins/app_service_rpc_plugin/test/app_service_plugin/app_service_app_extension_test.cc

```diff
@@ -0,0 +1,220 @@
+}
+
+TEST_F(AppServiceAppExtensionTest,
+       UnsubscribeFromAppService_UnsubscribeNotSubscribedAppServiceType_FAIL) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+
+  ASSERT_TRUE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType1));
+  ASSERT_EQ(1u, app_service_app_extension->Subscriptions().size());
+  ASSERT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+
+  EXPECT_FALSE(
+      app_service_app_extension->UnsubscribeFromAppService(kAppServiceType2));
+  EXPECT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+  EXPECT_EQ(1u, app_service_app_extension->Subscriptions().size());
+}
+
+TEST_F(AppServiceAppExtensionTest,
+       UnsubscribeFromAppService_UnsubscribeAll_SUCCESS) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+
+  for (const auto& app_service_type : {kAppServiceType1, kAppServiceType2}) {
+    ASSERT_TRUE(
+        app_service_app_extension->SubscribeToAppService(app_service_type));
+    ASSERT_TRUE(
+        app_service_app_extension->IsSubscribedToAppService(app_service_type));
+  }
+  ASSERT_EQ(2u, app_service_app_extension->Subscriptions().size());
+
+  app_service_app_extension->UnsubscribeFromAppService();
+
+  for (const auto& app_service_type : {kAppServiceType1, kAppServiceType2}) {
+    EXPECT_FALSE(
+        app_service_app_extension->IsSubscribedToAppService(app_service_type));
+  }
+  EXPECT_EQ(0u, app_service_app_extension->Subscriptions().size());
+}
+
+TEST_F(AppServiceAppExtensionTest, SaveResumptionData_SUCCESS) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+
+  ASSERT_TRUE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType1));
+  ASSERT_EQ(1u, app_service_app_extension->Subscriptions().size());
+  ASSERT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+
+  smart_objects::SmartObject resumption_data;
+  resumption_data[kResumptionDataKey] = "some resumption data";
+
+  app_service_app_extension->SaveResumptionData(resumption_data);
+
+  EXPECT_TRUE(resumption_data.keyExists(kResumptionDataKey));
+  EXPECT_TRUE(resumption_data.keyExists(kAppServiceInfoKey));
+  EXPECT_EQ(kAppServiceType1,
+            resumption_data[kAppServiceInfoKey][0].asString());
+}
+
+TEST_F(AppServiceAppExtensionTest, ProcessResumption_SUCCESS) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+  app_service_app_extension->UnsubscribeFromAppService();
+  ASSERT_EQ(0u, app_service_app_extension->Subscriptions().size());
+
+  smart_objects::SmartObject resumption_data;
+  resumption_data[kAppServiceInfoKey] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
```

Comment:
@YarikMamykin suggest to create a `SmartType_Array` object first, then push back data and then just assign this object to `kAppServiceInfoKey` field of `resumption_data`

Reply:
@AKalinich-Luxoft  fixed in 2ec9ea198cf54b563252051776bd08f48acb560f